package.json

```diff
@@ -18,7 +18,7 @@
   "dependencies": {
     "balanced-match": "0.1.0",
     "color": "^0.11.0",
-    "debug": "~0.7.4",
+    "debug": "3.1.0",
```

Comment:
ooof, that make nervous. I always pin to an exact version because I've been burned time and time again by upstream deps making breaking changes with incorrect semver. I trust no one!

But could go a different route on this project if you think it's best.

Reply:
debug is very widely used and I think it's more than safe to rely on it this way.
Otherwise you can go with ~3.1.0 but I think ^3 is ok

Reply:
Cool. Went with with `^`

Just curious on your projects do you decide what versions to let in based on each package? Like debug is stable, so pretty safe, but a less widely used one might get a tilde or pinned to a version?

Reply:
Today, I am using ^ everywhere + yarn (so yarn.lock) in order to get reproductible install and I don't have semver fail problems :)

Reply:
Ah! Gotcha, that makes sense. Thanks.